src/join.rs

```diff
@@ -113,6 +129,98 @@ impl<J: Join> std::iter::Iterator for JoinIter<J> {
     }
 }
 
+/// `JoinParIter` is a `ParallelIterator` over a group of `Storages`.
+#[must_use]
+pub struct JoinParIter<J>(J);
+
+impl<J> ParallelIterator for JoinParIter<J>
+  where
+    J: Join + Send,
+    J::Type: Send,
+    J::Value: Split + Send,
+    J::Mask: Send + Sync,
+{
+    type Item = J::Type;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+        where C: UnindexedConsumer<Self::Item>
+    {
+        let (keys, values) = self.0.open();
+        let producer = BitProducer((&keys).iter());
+        let values = UnsafeCell::new(values);
+        bridge_unindexed(JoinProducer::<J>::new(producer, &values), consumer)
+    }
+}
+
+struct JoinProducer<'a, J>
+  where
+    J: Join + Send,
```

Comment:
You'd still do

```rust
impl<'a, J>
```

but for

```rust
JoinProducer<'a, J::Mask, J::Value>
```

> How am going to call `Join::get` then? And how I am going to get `Join::Type`?

By adding these bounds only where they're necessary.